src/codegen/compile.ml

```diff
@@ -5824,7 +5824,12 @@ module MakeSerialization (Strm : Stream) = struct
           get_tag ^^ compile_eq_const Tagged.(int_of_tag ObjInd) ^^
           E.then_trap_with env "unvisited mutable data in serialize_go (ObjInd)" ^^
           get_tag ^^ compile_eq_const Tagged.(int_of_tag Array) ^^
-          E.then_trap_with env "unvisited mutable data in serialize_go (Array)" ^^
+            E.then_trap_with env "unvisited mutable data in serialize_go (Array)" ^^
```

Comment:
The difficulty here is that the tags have already been modified to encode the stream offset, conflicting with the range `>= ArraySliceMinimum`. Therefore, unfortunately, we cannot check whether the value is an original array slice tag or the stream offset but need to rely on `clear_array_slicing` been called at the beginning of the serialization.

Reply:
Oh wow, that's subtle.